src/cmds/tx.py

```diff
@@ -0,0 +1,190 @@
+# tx command imports
+import json
+from src.util.byte_types import hexstr_to_bytes
+
+# transaction imports
+from src.types.program import Program
+from src.types.sized_bytes import bytes32
+from src.types.coin_solution import CoinSolution
+from src.types.coin import Coin
+from src.util.ints import uint64
+from typing import List, Set
+
+from src.wallet.puzzles.puzzle_utils import (
+    make_assert_my_coin_id_condition,
+    make_assert_time_exceeds_condition,
+    make_assert_coin_consumed_condition,
+    make_create_coin_condition,
+    make_assert_fee_condition,
+)
+from src.wallet.puzzles.p2_delegated_puzzle import (
+    puzzle_for_pk,
+    solution_for_conditions,
+)
+
+
+# TODO: From wallet/wallet.py: refactor
+def make_solution(primaries=None, min_time=0, me=None, consumed=None, fee=0):
+    assert fee >= 0
+    condition_list = []
+    if primaries:
+        for primary in primaries:
+            condition_list.append(
+                make_create_coin_condition(primary["puzzlehash"], primary["amount"])
+            )
+    if consumed:
+        for coin in consumed:
+            condition_list.append(make_assert_coin_consumed_condition(coin))
+    if min_time > 0:
+        condition_list.append(make_assert_time_exceeds_condition(min_time))
+    if me:
+        condition_list.append(make_assert_my_coin_id_condition(me["id"]))
+    if fee:
+        condition_list.append(make_assert_fee_condition(fee))
+    print(condition_list)
+    return solution_for_conditions(condition_list)
+
+
+class SpendRequest:
```

Comment:
You can also use dataclasses, or even better Streamable class which also does type checking